Mathlib/SetTheory/Surreal/Basic.lean

```diff
@@ -439,6 +427,643 @@ lemma bddBelow_of_small (s : Set Surreal.{u}) [Small.{u} s] : BddBelow s := by
+  { rw [mul_option_neg_neg, lt_neg]
+    exact mul_option_lt_of_lt hy.neg (ih1_negy ihxy) (ih1_negx ihyx) j i l _ h }
+
+end
+
+/-- P1 follows from the induction hypothesis. -/
+theorem P1_of_ih : (x * y).Numeric := by
+  have ihxy := ih1xy ih
+  have ihyx := ih1yx ih
+  have ihxyn := ih1_negx (ih1_negy ihxy)
+  have ihyxn := ih1_negx (ih1_negy ihyx)
+  refine numeric_def.mpr ⟨?_, ?_, ?_⟩
+
+  · simp_rw [lt_iff_game_lt]
+    intro i
+    rw [rightMoves_mul_iff]
+    constructor
+    · intros j l
+      revert i
+      rw [leftMoves_mul_iff (GT.gt _)]
+      constructor
+      · intros i k
+        apply mul_option_lt hx hy ihxy ihyx
+      · simp only [← mul_option_symm (-y)]
+        rw [mul_option_neg_neg x]
+        have := mul_option_lt hy.neg hx.neg ihyxn ihxyn
+        simp [gt_iff_lt]
+        intros i j_1
+        exact this j_1 l i (toLeftMovesNeg (toRightMovesNeg j))
+    · intros j l
+      revert i
+      rw [leftMoves_mul_iff (GT.gt _)]
+      constructor
+      · simp only [← mul_option_symm y]
+        have := mul_option_lt hy hx ihyx ihxy
+        simp [gt_iff_lt]
+        intros i j_1
+        exact this j_1 l i j
+      · rw [mul_option_neg_neg y]
+        have := mul_option_lt hx.neg hy.neg ihxyn ihyxn
+        simp [gt_iff_lt]
+        intros i j_1
+        exact this i j j_1 (toLeftMovesNeg (toRightMovesNeg l))
+  all_goals (
+    cases x
+    cases y
+    rintro (⟨i,j⟩|⟨i,j⟩)
+  )
+  rw [mk_mul_moveLeft_inl]
+  pick_goal 2
+  rw [mk_mul_moveLeft_inr]
+  pick_goal 3
+  rw [mk_mul_moveRight_inl]
+  pick_goal 4
+  rw [mk_mul_moveRight_inr]
+  all_goals (
+    apply Numeric.sub
+    apply Numeric.add
+    apply ihnx ih
+    pick_goal 2
+    apply ihny ih
+    pick_goal 3
+    apply ihnxy ih
+  )
+  all_goals { solve_by_elim [IsOption.mk_left, IsOption.mk_right] }
+
+
+/-- A specialized induction hypothesis used to prove P2 and P4. -/
+def ih24 (x₁ x₂ y: PGame) : Prop :=
+  ∀ ⦃z⦄, (IsOption z x₁ → P24 z x₂ y) ∧ (IsOption z x₂ → P24 x₁ z y) ∧ (IsOption z y → P24 x₁ x₂ z)
+
+/-- A specialized induction hypothesis used to prove P4. -/
+def ih4 (x₁ x₂ y : PGame) : Prop :=
+  ∀ ⦃z w⦄, IsOption w y → (IsOption z x₁ → P2 z x₂ w) ∧ (IsOption z x₂ → P2 x₁ z w)
+
+/-! #### Specialize `ih'` to obtain specialized induction hypotheses for P2 and P4 -/
+
+variable (ih' : ∀ a, ArgsRel a (Args.P24 x₁ x₂ y) → P124 a)
+
+lemma ih₁₂ : ih24 x₁ x₂ y := by
+  rw [ih24]
+  refine (fun z => ⟨?_, ?_, ?_⟩) <;> (
+    refine fun h => ih' (Args.P24 _ _ _) (TransGen.single ?_)
+  )
+  · exact (cutExpand_add_right {y}).2 (cutExpand_pair_left h)
+  · exact (cutExpand_add_left {x₁}).2 (cutExpand_pair_left h)
+  · exact (cutExpand_add_left {x₁}).2 (cutExpand_pair_right h)
+
+
+lemma ih₂₁ : ih24 x₂ x₁ y := ih₁₂ (by
+  simp_rw [ArgsRel, InvImage, Args.toMultiset, Multiset.pair_comm] at ih' ⊢
+  suffices {x₁, y, x₂} = {x₂, y, x₁} by rwa [← this]
+  dsimp [← Multiset.singleton_add] at ih' ⊢
+  abel
+)
```

Comment:
I cleaned up some things in the next few lemmas following your other suggestions, but leaving this open in case you think there is more cleanup needed.